deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+
+    arg_parser.add_argument('-b', '--baseline', metavar='<file>', type=str, action='store',
+                            help='Baseline file to compare', required=True, dest='baseline_data')
+
+    arg_parser.add_argument('-f', '--file', metavar='<file>', type=str, action='store',
+                            help='Data file to compare', required=True, dest='data_source')
+    
+    return arg_parser.parse_args()
+
+
+def validate_parameters(parameters, datasource):
+    """Validate the input parameters
+
+    Args:
+        parameters (argparse.Namespace): script parameters.
+        datframe: data frame obtained from the data source file
+    """
+    # Check that the file exists
+    if not os.path.exists(parameters.baseline_data) or not os.path.exists(parameters.data_source):
+        print(f"The files provided do not exist")
+        sys.exit(1)
+    
+    # Check that the source file has more than 0 rows
+    dataframe = load_dataframe(datasource)
+    data_length = len(dataframe)    
+    if data_length == 0:
+        print(f"The source '{dataframe}' has not data rows or it has not CSV format")
+        sys.exit(1)
+
+
+def load_dataframe(datasource):
+    """Read the CSV and convert it to dataframe
+
+    Args:
+        datasource: data source file path.
+
+    Returns:
+        DataFrame: dataframe object.
+    """
+    return pd.read_csv(datasource)
+
+
+def calculate_basic_statistics(dataframe):
+    """Calculate basic statistics on the data source file for comparison.
+    
+    Args:
+        dataframe: Dataframe on which to calculate the statistics.
+    
+    Returns:
+        results: dict that contains all the statistics calculate for the dataframe
+    """
+    results = {}
+    daemons = dataframe['Daemon'].unique()
+
+    for daemon in daemons:
+        daemon_data = dataframe[dataframe['Daemon'] == daemon]
+        stats = []
+        for value in STATS_MAPPING.values():
+            stat = {
+                'Metric': value,
+                'Mean': round(float(daemon_data[value].mean()), 2),
+                'Median': round(float(daemon_data[value].median()), 2),
+                'Max value': round(float(daemon_data[value].max()), 2),
+                'Min value': round(float(daemon_data[value].min()), 2),
+                'Standard deviation': round(float(daemon_data[value].std()), 2),
+                'Variance': round(float(daemon_data[value].var()), 2)
+            }
+            stats.append(stat)
+        results[daemon] = stats
+    
+    return results
+
+
+def get_stat_value(metrics, value, stat):
+    """Checks if the metric is correct to return the specific statistic.
+
+    Args:
+        metrics: the metrics of a specific daemon.
+        value: value of this metric.
+        stat: statistics of this metric to be returned.
+    
+    Returns:
+        stat_value: value of the statistic desired.
+    """
+    for metric in metrics:
+        if metric['Metric'] == value:
+            return metric[stat]
```

Comment:
Is this function really necessary?

Reply:
No, it has been changed in https://github.com/wazuh/wazuh-qa/pull/5622/commits/26dc5cfa8d7f2339d13d5d5820d1ad7cceda3625